public/app/features/explore/utils/links.ts

```diff
@@ -113,6 +119,29 @@ export const getFieldLinksForExplore = (options: {
         if (!linkModel.title) {
           linkModel.title = getTitleFromHref(linkModel.href);
         }
+
+        // take over the onClick to report the click, then either call the original onClick or navigate to the URL
+        const origOnClick = linkModel.onClick;
+
+        linkModel.onClick = (...args) => {
+          reportInteraction('grafana_data_link_clicked', {
+            origin: link.origin || DataLinkConfigOrigin.Datasource,
+            app: CoreApp.Explore,
+            internal: false,
+          });
+
+          if (origOnClick) {
+            origOnClick?.apply(...args);
+          } else {
+            // for external links without an onClick, we want to duplicate default href behavior since onClick stops it
+            if (linkModel.target === '_blank') {
+              window.open(linkModel.href);
+            } else {
+              window.location.href = linkModel.href;
```

Comment:
Hmm... I guess I didn't think this through but now I wonder if tracking external links is actually feasible. If the URL is opened in the same window it will refresh the page probably before `reportInteraction` submits the data 🤔

Reply:
When I hit "preserve log" after running `window._debug.echo.enable()` it seemed to send the interaction event before the redirect.

Reply:
The echo service logs the input but I wonder if the request of `reportInteraction` will have time before it's canceled by the browser (unless it's using `navigator.sendBeacon()`, but I'm not sure how it works under the hood). Would be good to check it.

Reply:
I don't think there's a whole lot we can do here besides just attempt to report the click. I left a comment about the possible missing tracking data for external links that open in the same tab.